packages/build/tests/plugins/status/snapshots/tests.js.md

```diff
@@ -427,6 +545,274 @@ Generated by [AVA](https://ava.li).
     ␊
     (Netlify Build completed in 1ms)`
 
+## utils.status.show() implicit status is not used when an explicit call was made
+
+> Snapshot 1
+
+    `␊
+    ┌─────────────────────────────┐␊
+    │        Netlify Build        │␊
+    └─────────────────────────────┘␊
+    ␊
+    > Version␊
+      @netlify/build 1.0.0␊
+    ␊
+    > Flags␊
+      repositoryRoot: /file/path␊
+    ␊
+    > Current directory␊
+      /file/path␊
+    ␊
+    > Config file␊
+      /file/path␊
+    ␊
+    > Resolved config␊
+      plugins:␊
+        - package: /file/path␊
+    ␊
+    > Context␊
+      production␊
+    ␊
+    > Loading plugins␊
+       - /file/path␊
+    ␊
+    ┌─────────────────────────────────────┐␊
+    │ 1. onBuild command from /file/path │␊
+    └─────────────────────────────────────┘␊
+    ␊
+    ␊
+    (/file/path onBuild completed in 1ms)␊
+    ␊
+    ┌───────────────────────────────────────┐␊
+    │ 2. onSuccess command from /file/path │␊
+    └───────────────────────────────────────┘␊
+    ␊
+    ␊
+    (/file/path onSuccess completed in 1ms)␊
+    ␊
+    ┌─────────────────────────────┐␊
+    │           Summary           │␊
+    └─────────────────────────────┘␊
+    ␊
+    > Plugin /file/path ran successfully␊
+      summary␊
+    ␊
+    ┌─────────────────────────────┐␊
+    │   Netlify Build Complete    │␊
+    └─────────────────────────────┘␊
+    ␊
+    (Netlify Build completed in 1ms)`
+
+## utils.status.show() implicit status is not used when no call was made, with only onError
+
+> Snapshot 1
+
+    `␊
+    ┌─────────────────────────────┐␊
+    │        Netlify Build        │␊
+    └─────────────────────────────┘␊
+    ␊
+    > Version␊
+      @netlify/build 1.0.0␊
+    ␊
+    > Flags␊
+      repositoryRoot: /file/path␊
+    ␊
+    > Current directory␊
+      /file/path␊
+    ␊
+    > Config file␊
+      /file/path␊
+    ␊
+    > Resolved config␊
+      plugins:␊
+        - package: /file/path␊
+    ␊
+    > Context␊
+      production␊
+    ␊
+    > Loading plugins␊
+       - /file/path␊
+    ␊
+    ┌─────────────────────────────┐␊
+    │   Netlify Build Complete    │␊
+    └─────────────────────────────┘␊
+    ␊
+    (Netlify Build completed in 1ms)`
+
+## utils.status.show() implicit status is not used when there are no events
```

Comment:
I'm having a hard time understanding the difference between `implicit status is not used when there are no events`, `implicit status is not used when no call was made, with only onError` and `implicit status is used when no call was made`, `implicit status is used when no call was made, with only onEnd` and `implicit status is used when no events have made a call`.
How is the implicit status reflected in the snapshot? With previous tests I can see the summary, but the only differences I can see in these snapshots are the logs for specific commands.
What am I missing?

Reply:
Sorry this is confusing, and thanks for being thorough by checking the test snapshots!

Statuses bring together several features: build errors, plugin errors, build cancellation and plugin success statuses. Also we assign a default success status when the plugin does not do it explicitly. Finally, we need to ensure that:
  - this implicit success status is not overriding a build error or explicit call to `utils.status.show()`
  - `utils.status.show()` override any previous call to `utils.status.show()`
  - `utils.status.show()` cannot override a plugin or build error

This logic leads to the tests being rather complicated. I have written down some more detailed information below.

Reply:
Note that the specific tests you ask for are the last 3 ones in the post below.